### ISSUE TITLE
Semaphore Failed to authenticate than use network_cli module

### ISSUE THREAD

**Issue by arozhentsev** [ansible-semaphore/semaphore#696]
Hello!
Semaphore failed to authenticate than use network_cli module in playbook, but ansible run playbook whith no errors.

**Bad output from semaphore**:
`2:47:34 PM
Task 103 added to queue
2:47:36 PM
Preparing: 103
2:47:36 PM
access key rozhentsev installed
2:47:36 PM
Prepare task with template: cisco ip
2:47:36 PM
Updating repository git@172.16.175.71:administrator/cisco-set-ip-address.git
2:47:36 PM
* branch master -> FETCH_HEAD
2:47:36 PM
Из 172.16.175.71:administrator/cisco-set-ip-address
2:47:36 PM
Уже обновлено.
2:47:36 PM
access key rozhentsev installed
2:47:36 PM
No roles/requirements.yml file found. Skip galaxy install process.
2:47:36 PM
installing static inventory
2:47:41 PM
Started: 103
2:47:41 PM
Run task with template: cisco ip
2:47:42 PM
ansible-playbook 2.9.12
2:47:42 PM
configured module search path = ['/tmp/semaphore/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
2:47:42 PM
python version = 3.5.3 (default, Apr 5 2021, 09:00:41) [GCC 6.3.0 20170516]
2:47:42 PM
executable location = /usr/local/bin/ansible-playbook
2:47:42 PM
ansible python module location = /usr/local/lib/python3.5/dist-packages/ansible
2:47:42 PM
config file = /etc/ansible/ansible.cfg
2:47:42 PM
Using /etc/ansible/ansible.cfg as config file
2:47:42 PM
setting up inventory plugins
2:47:42 PM
host_list declined parsing /tmp/semaphore/inventory_103 as it did not pass its verify_file() method
2:47:42 PM
auto declined parsing /tmp/semaphore/inventory_103 as it did not pass its verify_file() method
2:47:42 PM
script declined parsing /tmp/semaphore/inventory_103 as it did not pass its verify_file() method
2:47:42 PM
Parsed /tmp/semaphore/inventory_103 inventory source with ini plugin
2:47:44 PM
Loading callback plugin default of type stdout, v2.0 from /usr/local/lib/python3.5/dist-packages/ansible/plugins/callback/default.py
2:47:44 PM
PLAYBOOK: playbook.yaml ********************************************************
2:47:44 PM
2:47:44 PM
tags: ('all',)
2:47:44 PM
Positional arguments: playbook.yaml
2:47:44 PM
connection: smart
2:47:44 PM
forks: 5
2:47:44 PM
remote_user: rozhentsev
2:47:44 PM
inventory: ('/tmp/semaphore/inventory_103',)
2:47:44 PM
timeout: 10
2:47:44 PM
become_method: sudo
2:47:44 PM
private_key_file: /itmh/tmp/semaphore/access_key_1
2:47:44 PM
verbosity: 4
2:47:44 PM
extra_vars: ('{"ANSIBLE_HOST_KEY_CHECKING": "False"}', ' interface=lo2', " ip='70.0.0.1 255.255.255.0'", ' description=blablabla2')
2:47:44 PM
1 plays in playbook.yaml
2:47:44 PM
PLAY [IP settings] *************************************************************
2:47:44 PM
2:47:44 PM
META: ran handlers
2:47:44 PM
TASK [Set Ip address] **********************************************************
2:47:44 PM
2:47:44 PM
task path: /itmh/tmp/semaphore/repository_2/playbook.yaml:8
2:47:44 PM
<172.16.88.56> using connection plugin network_cli
2:47:44 PM
<172.16.88.56> attempting to start connection
2:47:46 PM
<172.16.88.56> local domain socket does not exist, starting it
2:47:46 PM
<172.16.88.56> local domain socket listeners started successfully
2:47:46 PM
<172.16.88.56> local domain socket path is /itmh/tmp/semaphore/.ansible/pc/c13bf16a64
2:47:46 PM
<172.16.88.56> control socket path is /itmh/tmp/semaphore/.ansible/pc/c13bf16a64
2:47:46 PM
<172.16.88.56> loaded cliconf plugin ios from path /usr/local/lib/python3.5/dist-packages/ansible/plugins/cliconf/ios.py for network_os ios
2:47:46 PM
<172.16.88.56>
2:47:46 PM
<172.16.88.56> ESTABLISH LOCAL CONNECTION FOR USER: rozhentsev
2:47:46 PM
<172.16.88.56> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /tmp/semaphore/.ansible/tmp/ansible-local-26796fpdgayvp `"&& mkdir "` echo /tmp/semaphore/.ansible/tmp/ansible-local-26796fpdgayvp/ansible-tmp-1629452866.9853961-26818-175571397992161 `" && echo ansible-tmp-1629452866.9853961-26818-175571397992161="` echo /tmp/semaphore/.ansible/tmp/ansible-local-26796fpdgayvp/ansible-tmp-1629452866.9853961-26818-175571397992161 `" ) && sleep 0'
2:47:47 PM
<r1> Attempting python interpreter discovery
2:47:47 PM
<172.16.88.56> EXEC /bin/sh -c 'echo PLATFORM; uname; echo FOUND; command -v '"'"'/usr/bin/python'"'"'; command -v '"'"'python3.7'"'"'; command -v '"'"'python3.6'"'"'; command -v '"'"'python3.5'"'"'; command -v '"'"'python2.7'"'"'; command -v '"'"'python2.6'"'"'; command -v '"'"'/usr/libexec/platform-python'"'"'; command -v '"'"'/usr/bin/python3'"'"'; command -v '"'"'python'"'"'; echo ENDFOUND && sleep 0'
2:47:47 PM
<172.16.88.56> EXEC /bin/sh -c '/usr/bin/python && sleep 0'
2:47:47 PM
<r1> Python interpreter discovery fallback (unsupported Linux distribution: debian)
2:47:47 PM
Using module file /usr/local/lib/python3.5/dist-packages/ansible/modules/network/ios/ios_config.py
2:47:47 PM
<172.16.88.56> PUT /itmh/tmp/semaphore/.ansible/tmp/ansible-local-26796fpdgayvp/tmp160edf6p TO /itmh/tmp/semaphore/.ansible/tmp/ansible-local-26796fpdgayvp/ansible-tmp-1629452866.9853961-26818-175571397992161/AnsiballZ_ios_config.py
2:47:47 PM
<172.16.88.56> EXEC /bin/sh -c 'chmod u+x /tmp/semaphore/.ansible/tmp/ansible-local-26796fpdgayvp/ansible-tmp-1629452866.9853961-26818-175571397992161/ /tmp/semaphore/.ansible/tmp/ansible-local-26796fpdgayvp/ansible-tmp-1629452866.9853961-26818-175571397992161/AnsiballZ_ios_config.py && sleep 0'
2:47:47 PM
<172.16.88.56> EXEC /bin/sh -c '/usr/bin/python /tmp/semaphore/.ansible/tmp/ansible-local-26796fpdgayvp/ansible-tmp-1629452866.9853961-26818-175571397992161/AnsiballZ_ios_config.py && sleep 0'
2:48:12 PM
<172.16.88.56> EXEC /bin/sh -c 'rm -f -r /tmp/semaphore/.ansible/tmp/ansible-local-26796fpdgayvp/ansible-tmp-1629452866.9853961-26818-175571397992161/ > /dev/null 2>&1 && sleep 0'
2:48:12 PM
The full traceback is:
2:48:12 PM
at /usr/bin/python, but future installation of another Python interpreter could
2:48:12 PM
change this. See https://docs.ansible.com/ansible/2.9/reference_appendices/inte
2:48:12 PM
[WARNING]: Platform linux on host r1 is using the discovered Python interpreter
2:48:12 PM
WARNING: The below traceback may *not* be related to the actual failure.
2:48:12 PM
raise ConnectionError(to_text(msg, errors='surrogate_then_replace'), code=code)
2:48:12 PM
capabilities = Connection(module._socket_path).get_capabilities()
2:48:12 PM
rpreter_discovery.html for more information.
2:48:12 PM
File "/tmp/ansible_ios_config_payload_DFwFoI/ansible_ios_config_payload.zip/ansible/module_utils/network/ios/ios.py", line 86, in get_capabilities
2:48:12 PM
File "/tmp/ansible_ios_config_payload_DFwFoI/ansible_ios_config_payload.zip/ansible/module_utils/connection.py", line 185, in __rpc__
2:48:12 PM
fatal: [r1]: FAILED! => {
2:48:12 PM
"ansible_facts": {
2:48:12 PM
"module_args": {
2:48:12 PM
"discovered_interpreter_python": "/usr/bin/python"
2:48:12 PM
"backup": false,
2:48:12 PM
"after": null,
2:48:12 PM
"before": null,
2:48:12 PM
"defaults": false,
2:48:12 PM
"changed": false,
2:48:12 PM
"invocation": {
2:48:12 PM
"authorize": null,
2:48:12 PM
"auth_pass": null,
2:48:12 PM
"host": null,
2:48:12 PM
"backup_options": null,
2:48:12 PM
},
2:48:12 PM
"diff_ignore_lines": null,
2:48:12 PM
"diff_against": null,
2:48:12 PM
"intended_config": null,
2:48:12 PM
"provider": null,
2:48:12 PM
"description blablabla2",
2:48:12 PM
"lines": [
2:48:12 PM
],
2:48:12 PM
"ip address 70.0.0.1 255.255.255.0"
2:48:12 PM
"multiline_delimiter": "@",
2:48:12 PM
"match": "line",
2:48:12 PM
PLAY RECAP *********************************************************************
2:48:12 PM
2:48:12 PM
r1 : ok=0 changed=0 unreachable=0 failed=1 skipped=0 rescued=0 ignored=0
2:48:12 PM
2:48:12 PM
"parents": [
2:48:12 PM
"interface lo2"
2:48:12 PM
],
2:48:12 PM
"password": null,
2:48:12 PM
"running_config": null,
2:48:12 PM
}
2:48:12 PM
"ssh_keyfile": null,
2:48:12 PM
"timeout": null,
2:48:12 PM
"port": null,
2:48:12 PM
"replace": "line",
2:48:12 PM
"username": null
2:48:12 PM
"msg": "Failed to authenticate: Authentication failed."
2:48:12 PM
},
2:48:12 PM
}
2:48:12 PM
"src": null,
2:48:12 PM
"save_when": "never",
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
Exception ignored in: <function WeakValueDictionary.__init__.<locals>.remove at 0x7fec0b2e8268>
2:48:12 PM
Traceback (most recent call last):
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
File "/usr/lib/python3.5/weakref.py", line 117, in remove
2:48:12 PM
TypeError: 'NoneType' object is not callable
2:48:12 PM
Running playbook failed: exit status 2`

**Good output from ansible:**
`ansible-playbook 2.9.12
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/rozhentsev/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.5/dist-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 3.5.3 (default, Apr  5 2021, 09:00:41) [GCC 6.3.0 20170516]
Using /etc/ansible/ansible.cfg as config file
setting up inventory plugins
host_list declined parsing /itmh/home/rozhentsev/ansible/devnet/test/cisco-set-ip-address/hosts as it did not pass its verify_file() method
script declined parsing /itmh/home/rozhentsev/ansible/devnet/test/cisco-set-ip-address/hosts as it did not pass its verify_file() method
auto declined parsing /itmh/home/rozhentsev/ansible/devnet/test/cisco-set-ip-address/hosts as it did not pass its verify_file() method
Parsed /itmh/home/rozhentsev/ansible/devnet/test/cisco-set-ip-address/hosts inventory source with ini plugin
Loading callback plugin default of type stdout, v2.0 from /usr/local/lib/python3.5/dist-packages/ansible/plugins/callback/default.py

PLAYBOOK: playbook.yaml *******************************************************************************************************************************************************
Positional arguments: playbook.yaml
timeout: 10
forks: 5
connection: smart
verbosity: 4
become_method: sudo
tags: ('all',)
extra_vars: ('interface=lo2 ip="70.0.0.1 255.255.255.0" description=blablabla2',)
inventory: ('/itmh/home/rozhentsev/ansible/devnet/test/cisco-set-ip-address/hosts',)
1 plays in playbook.yaml

PLAY [IP settings] ************************************************************************************************************************************************************
META: ran handlers

TASK [Set Ip address] *********************************************************************************************************************************************************
task path: /itmh/home/rozhentsev/ansible/devnet/test/cisco-set-ip-address/playbook.yaml:8
<172.16.88.56> attempting to start connection
<172.16.88.56> using connection plugin network_cli
<172.16.88.56> local domain socket does not exist, starting it
<172.16.88.56> control socket path is /itmh/home/rozhentsev/.ansible/pc/68e22d4086
<172.16.88.56> local domain socket listeners started successfully
<172.16.88.56> loaded cliconf plugin ios from path /usr/local/lib/python3.5/dist-packages/ansible/plugins/cliconf/ios.py for network_os ios
<172.16.88.56>
<172.16.88.56> local domain socket path is /itmh/home/rozhentsev/.ansible/pc/68e22d4086
<172.16.88.56> ESTABLISH LOCAL CONNECTION FOR USER: rozhentsev
<172.16.88.56> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/rozhentsev/.ansible/tmp/ansible-local-852332y2xe52 `"&& mkdir "` echo /home/rozhentsev/.ansible/tmp/ansible-local-852332y2xe52/ansible-tmp-1629448208.4460044-8554-92950683761388 `" && echo ansible-tmp-1629448208.4460044-8554-92950683761388="` echo /home/rozhentsev/.ansible/tmp/ansible-local-852332y2xe52/ansible-tmp-1629448208.4460044-8554-92950683761388 `" ) && sleep 0'
<r1> Attempting python interpreter discovery
<172.16.88.56> EXEC /bin/sh -c 'echo PLATFORM; uname; echo FOUND; command -v '"'"'/usr/bin/python'"'"'; command -v '"'"'python3.7'"'"'; command -v '"'"'python3.6'"'"'; command -v '"'"'python3.5'"'"'; command -v '"'"'python2.7'"'"'; command -v '"'"'python2.6'"'"'; command -v '"'"'/usr/libexec/platform-python'"'"'; command -v '"'"'/usr/bin/python3'"'"'; command -v '"'"'python'"'"'; echo ENDFOUND && sleep 0'
<172.16.88.56> EXEC /bin/sh -c '/usr/bin/python && sleep 0'
<r1> Python interpreter discovery fallback (unsupported Linux distribution: debian)
Using module file /usr/local/lib/python3.5/dist-packages/ansible/modules/network/ios/ios_config.py
<172.16.88.56> PUT /itmh/home/rozhentsev/.ansible/tmp/ansible-local-852332y2xe52/tmp0wunmu5p TO /itmh/home/rozhentsev/.ansible/tmp/ansible-local-852332y2xe52/ansible-tmp-1629448208.4460044-8554-92950683761388/AnsiballZ_ios_config.py
<172.16.88.56> EXEC /bin/sh -c 'chmod u+x /home/rozhentsev/.ansible/tmp/ansible-local-852332y2xe52/ansible-tmp-1629448208.4460044-8554-92950683761388/ /home/rozhentsev/.ansible/tmp/ansible-local-852332y2xe52/ansible-tmp-1629448208.4460044-8554-92950683761388/AnsiballZ_ios_config.py && sleep 0'
<172.16.88.56> EXEC /bin/sh -c '/usr/bin/python /home/rozhentsev/.ansible/tmp/ansible-local-852332y2xe52/ansible-tmp-1629448208.4460044-8554-92950683761388/AnsiballZ_ios_config.py && sleep 0'
<172.16.88.56> EXEC /bin/sh -c 'rm -f -r /home/rozhentsev/.ansible/tmp/ansible-local-852332y2xe52/ansible-tmp-1629448208.4460044-8554-92950683761388/ > /dev/null 2>&1 && sleep 0'
[WARNING]: Platform linux on host r1 is using the discovered Python interpreter at /usr/bin/python, but future installation of another Python interpreter could change this.
See https://docs.ansible.com/ansible/2.9/reference_appendices/interpreter_discovery.html for more information.
changed: [r1] => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python"
    },
    "banners": {},
    "changed": true,
    "commands": [
        "interface lo2",
        "description blablabla2",
        "ip address 70.0.0.1 255.255.255.0"
    ],
    "invocation": {
        "module_args": {
            "after": null,
            "auth_pass": null,
            "authorize": null,
            "backup": false,
            "backup_options": null,
            "before": null,
            "defaults": false,
            "diff_against": null,
            "diff_ignore_lines": null,
            "host": null,
            "intended_config": null,
            "lines": [
                "description blablabla2",
                "ip address 70.0.0.1 255.255.255.0"
            ],
            "match": "line",
            "multiline_delimiter": "@",
            "parents": [
                "interface lo2"
            ],
            "password": null,
            "port": null,
            "provider": null,
            "replace": "line",
            "running_config": null,
            "save_when": "never",
            "src": null,
            "ssh_keyfile": null,
            "timeout": null,
            "username": null
        }
    },
    "updates": [
        "interface lo2",
        "description blablabla2",
        "ip address 70.0.0.1 255.255.255.0"
    ]
}

TASK [debug] ******************************************************************************************************************************************************************
task path: /itmh/home/rozhentsev/ansible/devnet/test/cisco-set-ip-address/playbook.yaml:17
<172.16.88.56> attempting to start connection
<172.16.88.56> using connection plugin network_cli
<172.16.88.56> found existing local domain socket, using it!
<172.16.88.56> ssh connection done, setting terminal
<172.16.88.56> loaded terminal plugin for network_os ios
<172.16.88.56> Response received, triggered 'persistent_buffer_read_timeout' timer of 0.1 seconds
<172.16.88.56> firing event: on_open_shell()
<172.16.88.56> Response received, triggered 'persistent_buffer_read_timeout' timer of 0.1 seconds
<172.16.88.56> Response received, triggered 'persistent_buffer_read_timeout' timer of 0.1 seconds
<172.16.88.56> Response received, triggered 'persistent_buffer_read_timeout' timer of 0.1 seconds
<172.16.88.56> firing event: on_become
<172.16.88.56> ssh connection has completed successfully
<172.16.88.56> Response received, triggered 'persistent_buffer_read_timeout' timer of 0.1 seconds
<172.16.88.56> Response received, triggered 'persistent_buffer_read_timeout' timer of 0.1 seconds
<172.16.88.56> updating play_context for connection
<172.16.88.56>
<172.16.88.56> local domain socket path is /itmh/home/rozhentsev/.ansible/pc/68e22d4086
ok: [r1] => {
    "msg": [
        "interface lo2",
        "description blablabla2",
        "ip address 70.0.0.1 255.255.255.0"
    ]
}

TASK [show interface config] **************************************************************************************************************************************************
task path: /itmh/home/rozhentsev/ansible/devnet/test/cisco-set-ip-address/playbook.yaml:21
<172.16.88.56> attempting to start connection
<172.16.88.56> using connection plugin network_cli
<172.16.88.56> found existing local domain socket, using it!
<172.16.88.56> updating play_context for connection
<172.16.88.56>
<172.16.88.56> local domain socket path is /itmh/home/rozhentsev/.ansible/pc/68e22d4086
<172.16.88.56> ESTABLISH LOCAL CONNECTION FOR USER: rozhentsev
<172.16.88.56> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/rozhentsev/.ansible/tmp/ansible-local-852332y2xe52 `"&& mkdir "` echo /home/rozhentsev/.ansible/tmp/ansible-local-852332y2xe52/ansible-tmp-1629448215.9200003-8647-52980453568747 `" && echo ansible-tmp-1629448215.9200003-8647-52980453568747="` echo /home/rozhentsev/.ansible/tmp/ansible-local-852332y2xe52/ansible-tmp-1629448215.9200003-8647-52980453568747 `" ) && sleep 0'
Using module file /usr/local/lib/python3.5/dist-packages/ansible/modules/network/ios/ios_command.py
<172.16.88.56> PUT /itmh/home/rozhentsev/.ansible/tmp/ansible-local-852332y2xe52/tmp6jgy0o4f TO /itmh/home/rozhentsev/.ansible/tmp/ansible-local-852332y2xe52/ansible-tmp-1629448215.9200003-8647-52980453568747/AnsiballZ_ios_command.py
<172.16.88.56> EXEC /bin/sh -c 'chmod u+x /home/rozhentsev/.ansible/tmp/ansible-local-852332y2xe52/ansible-tmp-1629448215.9200003-8647-52980453568747/ /home/rozhentsev/.ansible/tmp/ansible-local-852332y2xe52/ansible-tmp-1629448215.9200003-8647-52980453568747/AnsiballZ_ios_command.py && sleep 0'
<172.16.88.56> EXEC /bin/sh -c '/usr/bin/python /home/rozhentsev/.ansible/tmp/ansible-local-852332y2xe52/ansible-tmp-1629448215.9200003-8647-52980453568747/AnsiballZ_ios_command.py && sleep 0'
<172.16.88.56> EXEC /bin/sh -c 'rm -f -r /home/rozhentsev/.ansible/tmp/ansible-local-852332y2xe52/ansible-tmp-1629448215.9200003-8647-52980453568747/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
WARNING: The below traceback may *not* be related to the actual failure.
  File "/tmp/ansible_ios_command_payload_Z4Z2iL/ansible_ios_command_payload.zip/ansible/module_utils/network/ios/ios.py", line 135, in run_commands
    return connection.run_commands(commands=commands, check_rc=check_rc)
  File "/tmp/ansible_ios_command_payload_Z4Z2iL/ansible_ios_command_payload.zip/ansible/module_utils/connection.py", line 185, in __rpc__
    raise ConnectionError(to_text(msg, errors='surrogate_then_replace'), code=code)
fatal: [r1]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "auth_pass": null,
            "authorize": null,
            "commands": [
                "show running-config interface lo2"
            ],
            "host": null,
            "interval": 1,
            "match": "all",
            "password": null,
            "port": null,
            "provider": null,
            "retries": 10,
            "ssh_keyfile": null,
            "timeout": null,
            "username": null,
            "wait_for": null
        }
    },
    "msg": "show running-config interface lo2\r\nshow running-config interface lo2\r\n% Incomplete command.\r\n\r\nR1#"
}
`

**Comment by fiftin** [comment]
Can it be problem with python version? What python version do you use?

**Comment by arozhentsev** [comment]
Hello, fiftin and thank you for reply!
I do not think that the problem is in the python version, since both logs are from the same machine.
I am planning on using a semaphore to configure cisco ios. The problem is that I do not use ssh keys on ios devices. But you cannot create an inventory in a semaphore without an ssh key. So when the semaphore adds arguments to ansible-playbook, it adds --private-key. As a result, when connecting to the ios device, the publickey authorization method is selected. I don't know how to get around this. Looking at the source code, I found a condition in /api/tasks/runner.go

```
if t.inventory.SSHKeyID! = nil {
args = append (args, "--private-key =" + t.inventory.SSHKey.GetPath ())
}
```

Can you add a checkbox to the inventory UI "do not use a key" and an additional check to a condition like this one?

```
if (t.inventory.SSHKeyID! = nil and t.inventory.use_key = true) {
args = append (args, "--private-key =" + t.inventory.SSHKey.GetPath ())
}
```
Sorry for the syntax, I don't know golang.

**Comment by fiftin** [comment]
I added Access Key type None. You can use it for this purpose. Available from v2.7.8.

https://github.com/ansible-semaphore/semaphore/releases/tag/v2.7.8
